integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonPreAggregateRules.scala

```diff
@@ -806,67 +824,105 @@ case class CarbonPreAggregateQueryRules(sparkSession: SparkSession) extends Rule
     val updatedAggExp = aggregateExpressions.map {
```

Comment:
It seems lot of duplicate code, better do transform and use single  `case attr: AttributeReference` handle all here